Doc/c-api/refcounting.rst

```diff
@@ -97,6 +97,39 @@ objects.
    :c:func:`Py_DECREF`, and the same warning applies.
 
 
+.. c:function:: PyObject* _Py_Borrow(PyObject *o)
+
+   :term:`Borrow a reference <borrowed reference>` to an object: decrement the
+   object reference count and return the object.
+
+   The object must not be ``NULL``; if you aren't sure that it isn't
+   ``NULL``, use :c:func:`_Py_XBorrow`.
+
+   This function can be used as an expression to update an old C extension to
+   newer C API functions which return a :term:`strong reference`. For example,
+   replace ``frame->f_code`` with ``_Py_Borrow(PyFrame_GetCode(frame))``.
+
+   If the object can be destroyed before the last usage of the borrowed
+   reference, the :term:`borrowed reference` must be converted in-place to a
+   :term:`strong reference` by calling :c:func:`Py_INCREF`, or the
+   :c:func:`Py_NewRef` function can be used to create a new :term:`strong
+   reference`.
+
+   In other terms, the :c:func:`_Py_Borrow` function should be avoided whenever
```

Comment:
I'm wondering if the documentation should explain very briefly why borrowing references is bad? The documentation shouldn't go on a long rant, but at the moment the reader might be left wondering why `_Py_Borrow` should be avoided.

Reply:
The "term" syntax creates a link to: https://docs.python.org/dev/glossary.html#term-borrowed-reference which is where I explained in which cases borrowed references are unsafe.

I also wrote https://docs.python.org/dev/glossary.html#term-strong-reference which describes reference counting.